src/number_theory/pell.lean

```diff
@@ -183,6 +184,23 @@ begin
   exact lt_irrefl _ (((one_lt_sq_iff $ zero_le_one.trans ha.le).mpr ha).trans_eq prop),
 end
 
+/-- If a solution has `x > 1`, then `d` is positive. -/
+lemma d_pos_of_one_lt_x {a : solution₁ d} (ha : 1 < a.x) : 0 < d :=
```

Comment:
Naming lemmas after variable names (`d`) is usually frowned upon; though I don't have a great alternative here besides `pos_of_one_lt_x`

Reply:
OK, I can see why this is frowned upon, but leaving it out makes the name less descriptive. In the context of Pell's equation, `d` is a very standard name for the parameter, so absent a more convincing proposal, I would vote to leave these (four in total) names as they are.